overlord/snapstate/snapstate.go

```diff
@@ -885,7 +889,13 @@ func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
 		return false
 	}
 
+	// FIXME: make this more elegant
+	if si.Name() == "ubuntu-core" {
```

Comment:
if we fix CoreInfo, probably this can again switch from checking just type for core, to check whether the name is not the name of the "preferred" core snap